packages/nodes-base/nodes/WhatsApp/WhatsApp.node.ts

```diff
@@ -50,6 +61,53 @@ export class WhatsApp implements INodeType {
 			...mediaFields,
 			...messageTypeFields,
 			...mediaTypeFields,
+			...getSendAndWaitProperties([], 'message', undefined, {
+				noButtonStyle: true,
+				defaultApproveLabel: '✓ Approve',
+				defaultDisapproveLabel: '✗ Decline',
+			}).filter((p) => p.name !== 'subject'),
 		],
 	};
+
+	webhook = sendAndWaitWebhook;
+
+	customOperations = {
+		message: {
+			async [SEND_AND_WAIT_OPERATION](this: IExecuteFunctions) {
+				try {
+					const phoneNumberId = this.getNodeParameter('phoneNumberId', 0) as string;
+					const recipientPhoneNumber = (
+						this.getNodeParameter('recipientPhoneNumber', 0) as string
+					).replace(/[\-\(\)\+]/g, '');
+
+					const config = getSendAndWaitConfig(this);
+
+					const buttons = config.options.map((option) => {
+						return `*${option.label}:*\n_${config.url}?approved=${option.value}_\n\n`;
+					});
+
+					await this.helpers.requestWithAuthentication.call(this, WHATSAPP_CREDENTIALS_TYPE, {
+						baseURL: WHATSAPP_BASE_URL,
+						method: 'POST',
+						url: `${phoneNumberId}/messages`,
+						body: {
+							messaging_product: 'whatsapp',
+							text: {
+								body: `${config.message}\n\n${buttons.join('')}`,
```

Comment:
Can we extract the message builder into separate module and also include tests

Reply:
I do not see value of doing this - this is specific code for this specific case/operation, we will not be reusing it anywhere else
all code in this operation should be treated as separate unit and is tested by `packages/nodes-base/nodes/WhatsApp/node/sendAndWait.test.ts`